openwisp_controller/connection/tests/test_admin.py

```diff
@@ -0,0 +1,118 @@
+import json
+
+from django.test import TestCase
+from django.urls import reverse
+
+from ...config.models import Template
+from ...config.tests.test_admin import TestAdmin as TestConfigAdmin
+from ...tests.utils import TestAdminMixin
+from .. import settings as app_settings
+from ..models import Credentials, DeviceConnection, DeviceIp
+from .base import CreateConnectionsMixin, SshServerMixin
+
+
+class TestAdmin(TestAdminMixin, CreateConnectionsMixin,
+                SshServerMixin, TestCase):
+    template_model = Template
+    credentials_model = Credentials
+    deviceip_model = DeviceIp
+    connection_model = DeviceConnection
+    operator_permission_filters = [
+        {'codename__endswith': 'config'},
+        {'codename__endswith': 'device'},
+        {'codename__endswith': 'template'},
+        {'codename__endswith': 'connection'},
+        {'codename__endswith': 'credentials'},
+        {'codename__endswith': 'device_connection'},
+        {'codename__endswith': 'device_ip'},
+    ]
+    _device_params = TestConfigAdmin._device_params.copy()
+
+    def _get_device_params(self, org):
+        p = self._device_params.copy()
+        p['organization'] = org.pk
+        return p
+
+    def test_device_config_update(self):
```

Comment:
this test should be moved to `test_models` because we're not testing the admin here